src/sentry/eventstream/snuba.py

```diff
@@ -161,6 +161,27 @@ def insert(
         event_type = self._get_event_type(event)
         occurrence_data = self._get_occurrence_data(event)
 
+        # instead of normalizing and doing custom 'contexts' processing in snuba, we elect to do it here instead to
+        # avoid having to clutter up snuba code with business logic
+        if event_type == EventStreamEventType.Generic:
+            updated = False
+            if event_data.get("contexts"):
+                contexts = event_data.get("contexts", {})
```

Comment:
Nit: If we do `contexts = event_data.setdefault("contexts", {})` here, then we don't need to update this back into `event_data` at the end. Saves you needing to keep track of `updated`